rasterio/session.py

```diff
@@ -97,11 +98,17 @@ def cls_from_path(path):
 
         path = _parse_path(path)
 
+        aws_path_regex = r"a^"
+        rio_aws_s3_domains = os.getenv("RIO_AWS_S3_DOMAINS")
+        if rio_aws_s3_domains:
+            aws_path_regex = r"^https://.*(?:%s)" % "|".join(rio_aws_s3_domains.split(","))
```

Comment:
Thoughts about replacing "." with "\." in the domain names for regex escaping?

Reply:
yes doing that